contracts/libraries/LibIndexMap.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.9;
+
+struct IndexMap {
+  uint256[] inner;
+}
+
+using LibIndexMap for IndexMap global;
+
+library LibIndexMap {
```

Comment:
Write a thorough explanation of how the LibIndexMap works. Explain it in your own words.

Reply:
updated.